compiler/optimizer/RegisterCandidate.cpp

```diff
@@ -2249,7 +2249,7 @@ static void assign_candidate_loop_trace_increment(TR::Compilation *comp, TR_Regi
 bool
 TR_RegisterCandidates::assign(TR::Block ** cfgBlocks, int32_t numberOfBlocks, int32_t & lowestNumber, int32_t & highestNumber)
    {
-#if (defined(__IBMCPP__) || defined(__IBMC__))
+#if (defined(__IBMCPP__) || defined(__IBMC__)) && (!defined(__ibmxl__))
```

Comment:
Get rid of the extra parenthesis around `!defined` to keep it consistent with the rest of the code.